core/update.php

```diff
@@ -389,7 +389,9 @@ function update_task_list($set_active = NULL) {
     'finished' => 'Review log',
   );
 
-  return theme('task_list', array('items' => $tasks, 'active' => $active));
+  if (!user_is_anonymous()) {
+    return theme('task_list', array('items' => $tasks, 'active' => $active)); 
+  }
```

Comment:
This would make it so that task list is not shown on any of the pages of the update.php process when an anonymous user is completing updates with `$settings['update_free_access']` is set in settings.php.

Reply:
Good catch @quicksketch 👍 ...I will revise.